### PR TITLE
Fix verify-generated-swagger-docs script

### PR DESCRIPTION
**Pull request by cblecker**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
`verify-generated-swagger-docs` wasn't doing anything close to what it should have to verify that swagger docs weren't out of date. Most of these docs have moved out into staging/, and it was strictly focused on `pkg/`. The new script will look for the generated swagger files wherever they are, and diff them appropriately.

/cc @kubernetes/bash-firefighters @liggitt 

**Which issue(s) this PR fixes**:
ref: #74261, #74278

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: hack/verify-generated-swagger-docs.sh
Comment (liggitt) [inline hack/verify-generated-swagger-docs.sh]: is this as expensive as it looks?

Comment (cblecker) [reply hack/verify-generated-swagger-docs.sh]: Not really.. we use it in other places too where we want to copy and write the whole git tree for use in tests.

File: hack/update-generated-swagger-docs.sh
Comment (BenTheElder) [inline hack/update-generated-swagger-docs.sh]: this file LGTM

**Comment by BenTheElder** [comment]
this seems OK, I'd probably want to take a closer look at `verify-generated-swagger-docs.sh` ..

File: hack/verify-generated-swagger-docs.sh
Comment (liggitt) [reply hack/verify-generated-swagger-docs.sh]: ok. I'll defer to @BenTheElder for lgtm

**Comment by cblecker** [comment]
Ping @BenTheElder. This verify script is broken in it's current form.

/test all

File: hack/verify-generated-swagger-docs.sh
Comment (BenTheElder) [reply hack/verify-generated-swagger-docs.sh]: tar | untar is kinda expensive but one of the more reasonable options for copying to an isolated tempdir.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
thanks for bringing this back up my stack @cblecker and for the fix 👍

File: hack/verify-generated-swagger-docs.sh
Comment (BenTheElder) [reply hack/verify-generated-swagger-docs.sh]: (we also do this already as Christoph mentioned, it might be worth revisiting but I'd rather unblock for now)

**Pull request merged**
